### DISCUSSION TITLE
Does parent rewriting apply changes to branches and tags?

### DISCUSSION THREAD

**Discussion by GCheung55** [newren/git-filter-repo#287] [Q&A] [answered]
I'm working on removing some data from files in a projects history and I'm following the [docs on parent rewriting](https://htmlpreview.github.io/?https://github.com/newren/git-filter-repo/blob/docs/html/git-filter-repo.html#_parent_rewriting_2) to clean up some code. The project is several years old so it has many branches and tags to deal with.

I replaced `commit_A` with `commit_B` but still see `commit_A` in other branches. Is this expected behavior? Do I need to manually go to each branch to do the replacement?

The commands I used:
```
git replace $commit_A $commit_B
git filter-repo --force
```

**Comment by newren** [comment, marked as answer]
I believe
```
  git replace $commit_A $commit_B
```
is an error by default, because git-replace says "Unless -f is given, the replace reference must not yet exist.".  So that line is doing nothing.

Did you check that running
```
   git replace -l --format=medium
```
before running the filter-repo command shows the actual replacement setup?  Also, does
```
   git log $commit_A
```
actually give you the log message for commit_B?

If those are as expected, could you specify in more detail what you mean by "see[ing] commit_A in other branches"?  In particular, what are the git commands you are running that shows commit_A?  Was this after doing additional pushing/fetching, or seen in the repo where you ran the filtering operation before pushing/pulling it anywhere?
